lib/rules/no-extra-boolean-cast.js

```diff
@@ -63,6 +63,53 @@ module.exports = function(context) {
                     grandparent.callee.name === "Boolean")) {
                 context.report(node, "Redundant double negation in Boolean constructor call.");
             }
+        },
+        "CallExpression": function(node) {
+            var parent = node.parent;
+
+            if (node.callee.type !== "Identifier" || node.callee.name !== "Boolean") {
+                return;
+            }
+
+            // if (<bool>) ...
+            if (parent.type === "IfStatement") {
```

Comment:
The extra "in an if statement" part of the messages isn't necessary. It's good enough to say, "Redundant Boolean call." Then, you can squash all of these into a single statement. If you put the various node types into an array,  you can just use `indexOf` and a single `if`.